### PR TITLE
Improve ack-able message response logic and add tests

### PR DESCRIPTION
**Pull request by zackliu**

### CODE REVIEW

File: src/Microsoft.Azure.SignalR.Common/Utilities/AckHandler.cs [hunk 1]
```diff
@@ -63,8 +78,7 @@ private void CheckAcks()
                 {
                     if (_acks.TryRemove(pair.Key, out var ack))
                     {
-                        // If acks not coming back in time, do not throw an exception
-                        ack.Tcs.TrySetResult(false);
+                        ack.Tcs.TrySetCanceled();
```
Comment (vicancy) [inline src/Microsoft.Azure.SignalR.Common/Utilities/AckHandler.cs]: >TrySetCanceled [](start = 32, length = 14)

is `SetException` better?

File: src/Microsoft.Azure.SignalR.Common/Utilities/AckHandler.cs [hunk 2]
```diff
@@ -40,16 +44,27 @@ public AckHandler()
...
             if (_acks.TryRemove(id, out var ack))
             {
-                ack.Tcs.TrySetResult(isSuccess);
+                switch (ackStatus)
+                {
+                    case AckStatus.Ok:
+                        ack.Tcs.TrySetResult(true);
+                        break;
+                    case AckStatus.NotFound:
+                        ack.Tcs.TrySetResult(false);
```
Comment (vicancy) [inline src/Microsoft.Azure.SignalR.Common/Utilities/AckHandler.cs]: >TrySetResult [](start = 32, length = 12)

is SetException better?

File: src/Microsoft.Azure.SignalR.Common/Utilities/AckHandler.cs [hunk 3]
```diff
@@ -40,16 +44,27 @@ public AckHandler()
...
+                        break;
+                    case AckStatus.Timeout:
+                        ack.Tcs.TrySetCanceled();
```
Comment (vicancy) [inline src/Microsoft.Azure.SignalR.Common/Utilities/AckHandler.cs]: >TrySetCanceled [](start = 32, length = 14)

is SetException better?

Comment (zackliu) [reply src/Microsoft.Azure.SignalR.Common/Utilities/AckHandler.cs]: `TimeoutException` ?

File: src/Microsoft.Azure.SignalR.Common/Utilities/AckHandler.cs [hunk 4]
```diff
@@ -11,11 +11,15 @@ internal sealed class AckHandler : IDisposable
     {
         private readonly ConcurrentDictionary<int, AckInfo> _acks = new ConcurrentDictionary<int, AckInfo>();
         private readonly Timer _timer;
-        private readonly TimeSpan _ackInterval = TimeSpan.FromSeconds(5);
+        private readonly TimeSpan _ackInterval;
+        private readonly TimeSpan _ackTtl;
         private int _currentId = 0;
 
-        public AckHandler()
+        public AckHandler(int ackIntervalInMilliseconds = 5000, int ackTtlInMilliseconds = 10000)
```
Comment (vicancy) [inline src/Microsoft.Azure.SignalR.Common/Utilities/AckHandler.cs]: >5000 [](start = 58, length = 4)

can be less (or ttl need to be extended), 3 seconds?

File: src/Microsoft.Azure.SignalR.Common/ServiceConnections/ServiceConnectionContainerBase.cs [hunk 5]
```diff
@@ -258,7 +251,19 @@ public async Task<bool> WriteAckableMessageAsync(ServiceMessage serviceMessage,
...
+            {
+                case AckStatus.Ok:
+                    return true;
+                case AckStatus.NotFound:
+                    return false;
+                case AckStatus.Timeout:
+                    throw new TimeoutException($"Ack-able message {serviceMessage.GetType()} waiting for ack timed out.");
+                default:
+                    // should not be hit.
+                    return false;
```
Comment (vicancy) [inline src/Microsoft.Azure.SignalR.Common/ServiceConnections/ServiceConnectionContainerBase.cs]: >false [](start = 27, length = 5)

throw if should be hit

**Pull request merged**